core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -405,6 +408,34 @@ class AuthorizerIntegrationTest extends KafkaServerTestHarness {
   }
 
   @Test
+  def testPatternSubscriptionWithNoTopicAccess() {
+    addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), topicResource)
+    sendRecords(1, tp)
+    removeAllAcls()
+
+    addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Read)), groupResource)
+    try {
+      this.consumers.head.subscribe(Pattern.compile(topicPattern), new NoOpConsumerRebalanceListener)
+      consumeRecords(this.consumers.head)
+      Assert.fail("should have thrown exception")
+    } catch {
+      case e: TopicAuthorizationException => assertEquals(Collections.singleton(topic), e.unauthorizedTopics())
+    }
+  }
+
+  @Test
+  def testPatternSubscriptionWithTopicAndGroupRead() {
```

Comment:
It would be good to have a non-internal topic that doesn't match the pattern in this test as well. The reason is that we filter internal topics by default since 0.10.0.0 and we want the test to be robust if we end up changing where the internal topics filtering takes place. We could also consider having a test with internal topics filtering disabled.